### ISSUE TITLE
THREE.BufferGeometry.toJSON "Function.constructor.name" undefined for IE

### ISSUE THREAD

**Issue by bGute** [mrdoob/three.js#6349]
[Here](https://github.com/mrdoob/three.js/blob/dev/src/core/BufferGeometry.js#L979) the "Function.name" property is undefined for IE (see also [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name)). This will result in an error for line [44] (https://github.com/mrdoob/three.js/blob/dev/src/loaders/BufferGeometryLoader.js#L44) in the THREE.BufferGeometryLoader because attribute.type is undefined. 
Mabe you could do a fix like [here](http://stackoverflow.com/questions/6903762/function-name-not-supported-in-ie) until ECMAScript 6 is fully implemented for the IE.

**Comment by mrdoob** [comment]
Hmmm, I don't see what Function.name has to do with this...?

**Comment by bGute** [comment]
```type: javascript attribute.array.constructor.name ``` 
constructor.name is the concept of Function.name

**Comment by mrdoob** [comment]
Sorry, I'm still not understanding this.

**Comment by bGute** [comment]
Each object has a built-in property named "constructor" which refers to the function which setup the object. 
The name of the function is what we want to store in the json string e.g. Float32Array, so that we can deserialze it with the same parameters.
The point is that the line ```attribute.array.constructor.name  ``` return undefined when IE is used, because they do not support this property yet and i could not find out when they do so. Hence the question.
I hope it is more clear now...

**Comment by mrdoob** [comment]
Ah, yes. I understand now. So exporting `BufferGeometry` is broken in IE.

**Comment by dubejf** [comment]
What should we do about this?

1) Write a polyfill to patch the function prototype (like we do for `Math.sign`) ?

```
if ( Function.prototype.name === undefined && Object.defineProperty !== undefined ) {
    Object.defineProperty(Function.prototype, 'name', {
        get: function() {
            var name = this.toString().match(/^\s*function\s*(\S*)\s*\(/)[1];
            return name;
        }
    });
}
```

2) Instead of poly-filling `Function.prototype.name`, write a helper:
```
THREE.getFunctionName = function ( func ) {
	if ( Function.prototype.name === undefined ) {
		return func.toString().match(/^\s*function\s*(\S*)\s*\(/)[1];
	}
	return function.name;
}
```

and replace `attribute.array.constructor.name` with `THREE.getFunctionName( attribute.array.constructor )`.

I'm usually not a fan of polyfills, but this option is more intrusive.

3) Add a section to the wiki, explaining what the minimum requirements are for using three.js, and how to workaround some of the browser issues (see #3927 also). Let consumer of the library add the polyfill for themselves.

In this case, I feel like the first option is the best approach (and the least amount of work / support / surprise). What do you think?

**Comment by mrdoob** [comment]
Option 1 sounds good to me. As long as we add a comment as of why that is there.

**Comment by dubejf** [comment]
Fixed by #6771.
